File: Modules/Common/Techs/SIE.Web.Tech/Routings/Commands/ImportRouting.cs
using SIE.Security;
using SIE.Tech.Routings.ImportRoutings;
using SIE.Web.Command;
using SIE.Web.Common.Import.Commands;
using SIE.Web.Tech.Common.Commands;
using System;

namespace SIE.Web.MES.WorkOrders.Commands
{
    /// <summary>
    /// 工艺路线导入
    /// </summary>
    [JsCommand("SIE.Web.Tech.Routings.Commands.ImportRouting")]
    [AllowAnonymous]
    public class ImportRouting : ImportDataCommonCommand
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="importViewArgs"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ImportViewArgs importViewArgs, string scope)
        {
            return base.Excute(importViewArgs, "SIE.Tech.Routings.Routing,SIE.Tech");
        }

        /// <summary>
        /// 处理器
        /// </summary>
        /// <returns>类型</returns>
        protected override Type GetImportHandleType()
        {
            return typeof(ImportRoutingHandle);
        }

        /// <summary>
        /// 设置模板文件信息
        /// </summary>
        /// <param name="importViewArgs">importViewArgs</param>
        /// <param name="scope">scope</param>
        /// <returns>模板信息</returns>
        protected override TemplatePathInfo SetTemplateFileName(ImportViewArgs importViewArgs, string scope)
        {
            const string templateFileName = "工艺路线导入模板.xlsx";
            const string fileRelativePath = "Templates/" + templateFileName;
            return new TemplatePathInfo
            {
                FullPath = fileRelativePath,
                FileName = templateFileName
            };
        }
    }
}

File: Modules/Common/Techs/SIE.Web.Tech/Routings/Commands/ImportRouting.js
SIE.defineCommand('SIE.Web.Tech.Routings.Commands.ImportRouting', {
    extend: 'SIE.Web.Tech.Common.Commands.ImportDataCommonCommand',
    meta: { text: "导入", group: "edit", iconCls: "icon-Upload icon-blue" },    
    /*
    * 导入数据成功--刷新窗体
    * @param   {*} view 列表视图
    * @returns  *  grid
    * 默认执行成功后，刷新当前视图
    * 子类可以根据具体情况覆写1.自行处理指定刷新子列表，2.父列表或者所有tab子视图列表 3.调用自定义命令
    * 如果是视图使用了客制化查询命令，必须重写此方法，然后重新执行客制化命令中的方法
    * 示例：view._relations[0]._target.getCommands().items[0].方法名;
    */
    _importSuccess: function (view) {
        view.reloadData();
        view.routingControl.filterRoutingTreeData(view.routingControl, '');
    },
});